server/routers/account-recovery/generateAccountRecoveryChallenge.ts

```diff
@@ -45,18 +44,22 @@ export const generateAccountRecoveryChallenge = publicProcedure
       });
     }
 
-    const challengeValue = ChallengeUtils.generateChangeValue();
-    const challengeUpsertData = {
-      type: Config.CHALLENGE_TYPE,
+    if (!recoveryWallet.publicKey) {
+      throw new TRPCError({
+        code: "NOT_FOUND",
+        message: ErrorMessages.RECOVERY_MISSING_PUBLIC_KEY,
+      });
+    }
+
+    const challengeUpsertData = ChallengeUtils.generateChallengeUpsertData({
       purpose: ChallengePurpose.ACCOUNT_RECOVERY,
-      value: challengeValue,
-      version: Config.CHALLENGE_VERSION,
-      createdAt: new Date(),
+      publicKey: recoveryWallet.publicKey,
```

Comment:
We now need to load the `publicKey` when generating a challenge because `generateChallengeUpsertData` will generate a v1 (RSA) or v2 (EdDSA) challenge depending on the format of the public key.